newIDE/app/public/external/yarn/yarn-main.js

```diff
@@ -0,0 +1,75 @@
+import { createPathEditorHeader, fileExists } from '../utils/path-editor.js';
+
+const electron = require('electron');
+const electronWindow = electron.remote.getCurrentWindow();
+const ipcRenderer = electron.ipcRenderer;
+const fs = require('fs');
+const remote = electron.remote;
+
+let yarn = null;
+
+const saveAndClose = pathEditor => {
+  const savePath = pathEditor.state.fullPath;
+  yarn.data.saveTo(savePath, yarn.data.getSaveData('json'), () => {
+    ipcRenderer.send('yarn-changes-saved', savePath, null);
```

Comment:
The third argument (null) is never used, remove it entirely, otherwise it implies that it's used somewhere else for some purpose.